orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -843,3 +854,306 @@ def _update_onnx_model_initializers(self, state_tensors):
+                        {
+                            "step": step_tensor # int array of size 1
+                        }
+                    }
+                    "trainer_options":
+                    {
+                        "mixed_precision": mixed_precision_enabled_bool,
+                        "zero_stage": zero_enabled_stage,
+                        "world_rank": world_rank_int,
+                        "world_size": world_size_int,
+                        "optimizer_name": training_optimizer_name
+                    },
+                    "partition_info":
+                    {
+                        model_weight_name:
+                        {
+                            "original_dim": original_dimension_array
+                        }
+                    }
+                }
+            - when pytorch_format is true
+                {
+                    model_weight_name: model_weight_tensor
+                }
+
+        """
+        if not self._training_session:
+            warnings.warn("ONNX Runtime training session is not initialized yet. " \
+                          "Please run train_step or eval_step at least once before calling ORTTrainer.state_dict().", \
+                          UserWarning)
+            return self._load_state_dict.args[0] if self._load_state_dict else {}
+
+        states = {}
+
+        # load training session model states into the states dictionary
+        self._extract_model_states(states)
+        if pytorch_format:
+            # if pytorch_format is true, return a flat dictionary with only model states
+            # which is compatible with a PyTorch model
+            return states['model']['fp32']
+
+        # load training session optimizer states into the states dictionary
+        self._extract_optimizer_states(states)
+
+        # extract the relevant training configuration from the trainer and load them into the states dictionary
+        self._extract_trainer_options(states)
+
+        # add partition information in case of a distributed run
+        if self.options.distributed.deepspeed_zero_optimization.stage > 0:
+            partition_info = 'partition_info'
+            states[partition_info] = self._training_session.get_partition_info_map()
+
+        return states
+
+    def _load_model_states(self, training_session_state_dict, state_dict, strict):
+        """Load the model states onto the training session state dictionary"""
+
+        model = 'model'
+
+        # collect all initializer names from the current onnx graph
+        initializer_names = {node.name for node in self._onnx_model.graph.initializer}
+
+        # loaded_initializers dict will be loaded with all the model states from the state dictionary
+        # that are found in the initializer_names dictionary
+        loaded_initializers = {}
+
+        # create an entry for the model in the training session state dictionary
+        if model not in training_session_state_dict:
```

Comment:
what is the reason for maintaining the 'model' in training_session_state_dict

Reply:
My intention is that if in the future we support fp16 states being copied directly instead of extending the fp32 weights to fp16 weights, we need a way to communicate this information to the backend.
Maybe we can target this work as and when required (in which case, I can remove it for now). What are your thoughts on this?

Reply:
it should be simple enough to extend it to include model states later when there is a need to do so. For now, we can skip including this as it is unused.

Reply:
ok sure. I can remove this for now then.